File: SQL-AI/ProgrammingLanguage/Leetcode-Implementation/Python/TwoSum.py
class Solution:
    # @return a tuple, (index1, index2)
    def twoSum(self, num, target):
        """ two number add up to target"""
        # non-zero based
        visited = dict()
        for idx in range(len(num)):
            if (target - num[idx]) in visited:
                return (visited[target-num[idx]] + 1, idx+1)
            
            visited[num[idx]] = idx

File: SQL-AI/ProgrammingLanguage/Leetcode-Implementation/Ruby/two_sum.rb
# Input: numbers={2, 7, 11, 15}, target=9
# Output: index1=1, index2=2

# @param {Integer[]} nums
# @param {Integer} target
# @return {Integer[]}
def two_sum(nums, target)

  number_position = {}
  nums.each_with_index do |number, index|
    if number_position[target - number] != nil
      return [number_position[target - number] + 1, index + 1]
    else
      number_position[number] = index
    end
  end

  return [0, 0] # indicate failure
end

puts two_sum([2, 7, 11, 15], 9).inspect
